tests/services/analysis_starter/test_nextflow_file_creation.py

```diff
@@ -100,34 +109,61 @@ def test_nextflow_config_file_content(
     assert content.rstrip() == expected_content.rstrip()
 
 
-@pytest.mark.parametrize(
-    "file_creator_fixture, case_id_fixture, expected_content_fixture",
-    [
-        (
-            "raredisease_sample_sheet_creator",
-            "raredisease_case_id",
-            "raredisease_sample_sheet_expected_content",
-        )
-    ],
-    ids=["raredisease"],
-)
-def test_nextflow_sample_sheet_content(
-    file_creator_fixture: str,
-    case_id_fixture: str,
-    expected_content_fixture: str,
-    request: pytest.FixtureRequest,
+@pytest.fixture
+def sample_sheet_scenario(
+    mocker: MockerFixture,
+    raredisease_sample_sheet_creator2: RarediseaseSampleSheetCreator,
+    raredisease_sample_sheet_expected_content: list[list[str]],
+    rnafusion_sample_sheet_creator: RNAFusionSampleSheetCreator,
+    rnafusion_sample_sheet_content_list: list[list[str]],
+) -> dict:
+    return {
+        Workflow.RAREDISEASE: (
+            raredisease_sample_sheet_creator2,
+            raredisease_sample_sheet_expected_content,
+            mocker.patch.object(raredisease, "write_csv"),
+        ),
+        Workflow.RNAFUSION: (
+            rnafusion_sample_sheet_creator,
+            rnafusion_sample_sheet_content_list,
+            mocker.patch.object(rnafusion, "write_csv"),
+        ),
+    }
+
+
+@pytest.mark.parametrize("workflow", [Workflow.RAREDISEASE, Workflow.RNAFUSION])
+def test_nextflow_sample_sheet_creators(
+    mocker: MockerFixture,
+    nextflow_case_id: str,
+    nextflow_case_path: Path,
+    sample_sheet_scenario: dict,
+    workflow: Workflow,
 ):
-    """Test that the sample sheet content is created correctly."""
-    # GIVEN a sample sheet content creator, a case id and a case path
-    file_creator: NextflowSampleSheetCreator = request.getfixturevalue(file_creator_fixture)
-    case_id: str = request.getfixturevalue(case_id_fixture)
+    # GIVEN a sample sheet creator, an expected output and a mocked file writer
+    sample_sheet_creator, expected_content, write_mock = sample_sheet_scenario[workflow]
```

Comment:
If we move this fixture to anothe rmodule it will be better to be explicit on which writter we are mocking
```suggestion
    sample_sheet_creator, expected_content, write_csv_mock = sample_sheet_scenario[workflow]
```

Reply:
and I think as the test relies on assertion over it, it is better to make clear what we are talking about

Reply:
Discussed over slack, to be continued in another PR